src/baseline/expr.rs

```diff
@@ -1282,13 +1286,26 @@ where
     }
 
     fn emit_intrinsic_assert(&mut self, e: &'ast ExprCallType, _: Reg) {
-        let lbl_div = self.asm.create_label();
+        // throw Error if assertion failed
+        let lbl_assert = self.asm.create_label();
         self.emit_expr(&e.args[0], REG_RESULT.into());
 
         self.asm.emit_comment(Comment::Lit("check assert"));
         self.asm
-            .test_and_jump_if(CondCode::Zero, REG_RESULT, lbl_div);
-        self.asm.emit_bailout(lbl_div, Trap::ASSERT, e.pos);
+            .test_and_jump_if(CondCode::NonZero, REG_RESULT, lbl_assert);
+        let csite = self.jit_info.map_csites.get(e.id).unwrap().clone();
+
+        match csite.args.get(1).unwrap() {
+            Arg::Stack(offset, _, _) => {
+                self.emit_lit_str_value(&"assert failed".to_string(), REG_RESULT);
+                self.asm
+                    .store_mem(MachineMode::Ptr, Mem::Local(*offset), REG_RESULT.into());
+            }
+            _ => panic!("unexpected argument for assert"),
+        };
+        self.emit_call_site(&csite, e.pos, REG_RESULT.into());
+        self.asm.throw(REG_RESULT, e.pos);
```

Comment:
At least in the new baseline compiler we should emit this code out-of-line. However you could already implement this as slow path in `baseline/asm.rs`. Both implementations should be able to share the same code so it's not additional work.

Reply:
Maybe I'm thinking to complicated, but if I implement it as slow path, I would to add/move a lot of functions, which I call for creating the object to asm, e.g. the `emit_lit_str_value` and `emit_call_site`. For now I will leave it as it is, but revisit it, when I'm working on the new baseline compiler.